src/Features/CSharp/Portable/UsePatternMatching/CSharpIsAndCastCheckWithoutNameDiagnosticAnalyzer.cs

```diff
@@ -215,7 +215,9 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext context)
 
             var updatedCompilation = semanticModel.Compilation.ReplaceSyntaxTree(
                 semanticModel.SyntaxTree, updatedSyntaxTree);
+#pragma warning disable RS1030 // Do not invoke Compilation.GetSemanticModel() method within a diagnostic analyzer
             return updatedCompilation.GetSemanticModel(updatedSyntaxTree);
```

Comment:
I believe this analyzer needs to be rewritten. I was unable to move this to shared layer as it seems to be using Workspace APIs to update trees and compilation and such, which seems bad for analyzer performance. It is fine to suppress this for now but this certainly needs to be rewritten.

Reply:
>I believe this analyzer needs to be rewritten. I was unable to move this to shared layer as it seems to be using Workspace APIs to update trees and compilation and such

It shouldn't be.  Can you point out what you mean here?  The only workspace APi it uses it to get an option.  It also uses SyntaxEditor/Generator, but those aren't workspace dependent.

> which seems bad for analyzer performance. 

Potentially, but forking trees/semantic-models has been something supported and not-discouraged afaict.  

Def remove if you want, but we may need some new way to answer the questions this particular anlyzer needs.